libraries/lib-utility/MathApprox.h

```diff
@@ -0,0 +1,45 @@
+/*  SPDX-License-Identifier: GPL-2.0-or-later */
+/*!********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  MathApprox.h
+
+  Matthieu Hodgkinson
+
+**********************************************************************/
+#pragma once
+
+#include <cstdint>
+
+/*!
+ * @brief Approximates the base-2 logarithm of a float to two decimal places,
+ * adapted from https://stackoverflow.com/a/28730362
+ *
+ * @details Accuracy claim is demonstated in MathApproxTest.cpp, and coincides
+ * with @Paul-Licameli's analysis (copied from conversation):
+ * "If x == 1.0f, then log2 before the += correction actually becomes -1 --
```

Comment:
@Paul-Licameli would you be so kind to review that me pasting your Discord message here is relevant?